pkg/tasks/common.go

```diff
@@ -0,0 +1,100 @@
+/*
+Copyright 2021 The KubeOne Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tasks
+
+import (
+	"bytes"
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	kubeletconfigv1beta1 "k8s.io/kubelet/config/v1beta1"
+	"sigs.k8s.io/yaml"
+)
+
+const (
+	kubeadmEnvFlagsFile   = "/var/lib/kubelet/kubeadm-flags.env"
+	kubeletKubeadmArgsEnv = "KUBELET_KUBEADM_ARGS"
+	kubeletConfigFile     = "/var/lib/kubelet/config.yaml"
+)
+
+func unmarshalKubeletFlags(buf []byte) (map[string]string, error) {
+	// throw away KUBELET_KUBEADM_ARGS=
+	s1 := strings.SplitN(string(buf), "=", 2)
+	if len(s1) != 2 {
+		return nil, errors.New("can't parse: wrong split length")
+	}
+
+	envValue := strings.Trim(s1[1], `"`)
+	flagsvalues := strings.Split(envValue, " ")
+	kubeletflagsMap := map[string]string{}
+
+	for _, flg := range flagsvalues {
+		fl := strings.Split(flg, "=")
+		if len(fl) != 2 {
+			return nil, errors.New("wrong split length")
+		}
+		kubeletflagsMap[fl[0]] = fl[1]
+	}
+
+	return kubeletflagsMap, nil
+}
+
+func marshalKubeletFlags(kubeletflags map[string]string) []byte {
+	kvpairs := []string{}
+	for k, v := range kubeletflags {
+		kvpairs = append(kvpairs, fmt.Sprintf("%s=%s", k, v))
+	}
+
+	sort.Strings(kvpairs)
+
+	var buf bytes.Buffer
+	fmt.Fprintf(&buf, `%s="`, kubeletKubeadmArgsEnv)
+
+	for i, val := range kvpairs {
+		format := "%s "
+		if i == len(kvpairs)-1 {
+			format = "%s"
+		}
+		fmt.Fprintf(&buf, format, val)
+	}
```

Comment:
Is this the same as `strings.Join(kvpairs, " ")`? If so, this can be reduced to

```
return fmt.Sprintf(`%s="%s"`, kubeletKubeadmArgsEnv, strings.Join(kvpairs, " "))
```